### ISSUE TITLE
Support for building with LuaJIT with cmake

### ISSUE THREAD

**Issue by Asmageddon** [ivansafrin/Polycode#635]
As far as I'm aware, LuaJIT already works with Polycode(and well, there's no reason why it shouldn't), but as it stands, it requires manually replacing Lua5.1 with LuaJIT, and as such, I would like to propose/request the ability to build LuaJIT instead of Lua5.1 using cmake. I'd gladly fix it up myself, but unfortunately, I am not familiar with cmake, and truth be told, the whole system seems extremely cryptic to me.

**Comment by ivansafrin** [comment]
Yeah, it's a good idea. Non-cmake projects can be a pain to setup with cmake's automated build stuff, so i've procrastinated doing this, but it should def. happen.

**Comment by Asmageddon** [comment]
Speaking of CMake, I have not worked with C/C++ a whole lot, but surely there must be some... more modern... equivalent, that could replace it? From the few glimpses into how CMake works, and its various files, I've only barely managed to keep my sanity, while coming to the conclusion that it certainly simply cannot be even remotely close to the best solution for building.

**Comment by ivansafrin** [comment]
I really really wish there was a better alternative to CMake as it gives me the heebie jeebies myself, but so far it seems to be the only option that gets the job done as far as I can tell. If you find a better alternative, I would always be up for trying it!

**Comment by fodinabor** [comment]
https://github.com/LuaDist/luajit
I don't know where this cmake folder comes from (it's not in the normal repo of LuaJIT).. but I'd try it, what do you think?

**Comment by lammermann** [comment]
I think [premake](https://premake.github.io/) is a good alternative to cmake. It seems to come with a much cleaner syntax (lua based) and the generated buildfiles are cleaner then those of cmake too.
Both bullet and box2d use it as there build tool.
